modules/imaging_uploader/jsx/UploadForm.js

```diff
@@ -63,7 +63,7 @@ class UploadForm extends Component {
     formData[field] = value;
 
     if (field === 'mriFile') {
-      if (value.name !== '' && formData.IsPhantom === 'N') {
+      if (value.name && value.name !== '' && formData.IsPhantom === 'N') {
```

Comment:
Not sure why this line changed.
The `value.name `check will return false if value.name is an empty string. So `value.name !== ''` would cover that check. Also if the submit button is pressed when "File to upload" has not been done, the browser will request the user to Select a file. The check on `value.name` will return false if it's an empty string. I'm also unsure if either the `value.name` or `value.name !== ''` are needed at all because of the browser check when the `File to Upload` field is required.

Reply:
i added `value.name` to the condition to get rid of an undefined variable error that popped up in the error log. i can't seem to find the screen shot to show you though

Reply:
@zaliqarosli I haven't been able to replicate that error case. I sort of think the two checks are unnecessary but maybe they somehow can be thrown. I just thought to comment on it. It doesn't really matter.

Reply:
By looking at the code, my best guess is that the error was thrown when `value.name` was `null` but then ` value.name.replace()` was called on an undefined variable. This would probably happen when `componentDidMount` is called since it called onFormChange like this: `this.onFormChange(this.state.form.IsPhantom.name, null);`

Not sure if the best solution is to add this if statement, or to address the fact it's passing `null` as the value.

But I agree with @maltheism, if there's the `if (value.name)` check then the `value.name !== ''` becomes redundant.

Reply:
if i remove `value.name !== ''`, would either of you be willing to do another test in case that breaks something? @HenriRabalais @maltheism

Reply:
@zaliqarosli I can test again.